backend/de.metas.util/src/main/java/de/metas/util/time/DurationUtils.java

```diff
@@ -35,10 +38,32 @@
 @UtilityClass
 public class DurationUtils
 {
-	public static Duration toDuration(@NonNull final BigDecimal durationBD, @NonNull final TemporalUnit unit)
+	private static final List<TemporalUnit> supportedTemporalUnits = ImmutableList.of(ChronoUnit.YEARS, ChronoUnit.MONTHS, ChronoUnit.DAYS, ChronoUnit.HOURS, ChronoUnit.MINUTES, ChronoUnit.SECONDS, ChronoUnit.NANOS);
+
+	public static Duration toWorkDuration(final @NonNull BigDecimal durationBD, final @NonNull TemporalUnit unit)
+	{
+		BigDecimal duration = durationBD;
+		TemporalUnit currentUnit = unit;
+		long durationLong = -1;
+		while (durationLong < 0)
+		{
+			try
+			{
+				durationLong = duration.longValueExact();
+			}
+			catch (final ArithmeticException ae)
```

Comment:
don't lean on exception. instead (found in `de.metas.util.NumberUtils#stripTrailingDecimalZeros`), i think you can do 
`stripTrailingZeros()` and then check the scale

Reply:
changed.